### PR TITLE
Update ArborX for new Kokkos

### PR DESCRIPTION
**Pull request by aprokop**

### CODE REVIEW

File: var/spack/repos/builtin/packages/arborx/package.py [hunk 1]
```diff
@@ -9,42 +9,61 @@
...
+        'hip': (False,  "enable HIP backend")
+    }
 
-    variant('cuda', default=False, description='enable Cuda backend')
-    variant('openmp', default=False, description='enable OpenMP backend')
-    variant('serial', default=True, description='enable Serial backend (default)')
     variant('mpi', default=True, description='enable MPI')
+    for backend in kokkos_backends:
+        deflt, descr = kokkos_backends[backend]
+        variant(backend.lower(), default=deflt, description=descr)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/arborx/package.py:34]: You could also use a single multi-valued variant here, not sure what your preference is but it doesn't matter to me

**Review by adamjstewart — COMMENTED** [review]
I see that you've contributed to arborx upstream. Want to be listed as an official maintainer?

**Comment by aprokop** [comment]
> I see that you've contributed to arborx upstream. Want to be listed as an official maintainer?

Done. Thanks for the review.

File: var/spack/repos/builtin/packages/arborx/package.py (hunk 1 above)
Comment (aprokop) [reply var/spack/repos/builtin/packages/arborx/package.py:34]: Lets leave it like this for now to be consistent with `kokkos-kernels`.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**